sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRegistryRecordedTestSanitizer.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Web;
+using Azure.Core.TestFramework;
+
+namespace Azure.Containers.ContainerRegistry.Tests
+{
+    public class ContainerRegistryRecordedTestSanitizer : RecordedTestSanitizer
+    {
+        public List<string> FormEncodedBodySanitizers { get; } = new List<string>();
+
+        public ContainerRegistryRecordedTestSanitizer()
+            : base()
+        {
+            JsonPathSanitizers.Add("$..access_token");
+            JsonPathSanitizers.Add("$..refresh_token");
+
+            FormEncodedBodySanitizers.Add("access_token");
+            FormEncodedBodySanitizers.Add("refresh_token");
+        }
+
+        public override string SanitizeTextBody(string contentType, string body)
+        {
+            string jsonSanitizedBody = base.SanitizeTextBody(contentType, body);
+
+            if (FormEncodedBodySanitizers.Count == 0)
+            {
+                return jsonSanitizedBody;
+            }
+
+            try
+            {
+                if (contentType == "application/x-www-form-urlencoded")
+                {
+                    string urlEncodedSanitizedBody = string.Empty;
+                    NameValueCollection queryParams = HttpUtility.ParseQueryString(jsonSanitizedBody);
```

Comment:
It might be worth editing the queryParams directly and calling .ToString() on it to get UrlFormattedValue back.

Reply:
Ah, magic!  Didn't know that existed - thanks!